vault/expiration.go

```diff
@@ -422,6 +480,88 @@ func (m *ExpirationManager) Restore(errorFunc func()) (retErr error) {
 	m.restoreModeLock.Unlock()
 
 	m.logger.Info("lease restore complete")
+
+	if atomic.LoadUint32(&m.SHA1HashedLeasesCleared) == 0 {
```

Comment:
We probably want to run this even if no sha1 hashes don't exist so we can guarantee the root tokens will be upgraded regardless. Can we check `SHA1HashedUnleasedTokensUpgraded` here instead?